src/nomad_material_processing/general/schema.py

```diff
@@ -0,0 +1,278 @@
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from nomad.datamodel.datamodel import (
+        EntryArchive,
+    )
+    from structlog.stdlib import (
+        BoundLogger,
+    )
+
+from nomad.config import config
+from nomad.datamodel.data import Schema
+from nomad.datamodel.metainfo.annotations import ELNAnnotation, ELNComponentEnum
+from nomad.metainfo import Quantity, SchemaPackage
+
+import numpy as np
+import yaml
+import json
+import math
+
+from nomad.datamodel.data import EntryData, EntryDataCategory, ArchiveSection
+
+from nomad.metainfo import (
+    SchemaPackage,
+    Package,
+    Quantity,
+    SubSection,
+    Datetime,
+    Section,
+    Category,
+    MEnum,
+    Reference,
+)
+
+from nomad.datamodel.metainfo.basesections import (
+    CompositeSystem,
+    Component,
+    System,
+    Activity,
+    ActivityStep,
+    ProcessStep,
+    Process,
+    CompositeSystemReference,
+    PureSubstanceSection,
+    PureSubstanceComponent,
+)
+from nomad.config import config
+from nomad.datamodel.metainfo.annotations import (
+    ELNAnnotation,
+    ELNComponentEnum,
+)
+
+from nomad_material_processing import (
+    SubstrateReference,
+    Geometry,
+)
+
+
+class Recipe(ArchiveSection):
+    """
+    A Recipe for a material processing experiment.
+
+    This class will be subclassed for each process that needs a recipe.
+
+    The subclass will inherit Recipe and a specific Process class.
+
+    The only difference between the Recipe and the actual Process is that
+    the datetime and the input samples Entities are hidden in the Recipe.
+    """
+
+    pass
+
+
+class EtchingStep(ActivityStep):
+    """
+    A step of etching process.
+    """
+
+    m_def = Section()
+    duration = Quantity(
+        type=np.float64,
+        description='The elapsed time since the annealing process started.',
+        a_eln=ELNAnnotation(
+            component=ELNComponentEnum.NumberEditQuantity, defaultDisplayUnit='minute'
+        ),
+        unit='second',
+    )
+    temperature = Quantity(
+        type=np.float64,
+        description='The temperature of the etching process.',
+        a_eln={'component': 'NumberEditQuantity', 'defaultDisplayUnit': 'celsius'},
+        unit='celsius',
+    )
+    agitation = Quantity(
+        type=MEnum(
+            'Magnetic Stirring',
+            'Sonication',
+        ),
+        description='The agitation method used during the etching process.',
+        a_eln=ELNAnnotation(
+            component=ELNComponentEnum.EnumEditQuantity,
+        ),
+    )
+    etching_reagents = SubSection(section_def=CompositeSystem, repeats=True)
+
+
+class Etching(Process, EntryData):
+    """
+    Class autogenerated from yaml schema.
```

Comment:
Maybe just add a short description. Otherwise we will forget it and it will remain like this for ever...
```suggestion
    The process of chemically removing a thin layer from a sample surface, usually to leave behind a pattern.
```